### ISSUE TITLE
Add Intellihide option to hide the panel when a window of any size is opened

### ISSUE THREAD

**Issue by yioannides** [home-sweet-gnome/dash-to-panel#2223]
**Describe the bug**
When Panel Intellihide is activated, the [The panel hides from "All windows"] entry does not hide the panel when any window is open. (it will hide for maximized windows though, as intended)

**Linux distribution and version**
Fedora 41

**GNOME Shell version**
47.2

**Dash-to-Panel version**
64

**Where was Dash-to-Panel installed from?**
Restored from cloud / updated via Extension Manager

**Screenshots / Video captures**
![image](https://github.com/user-attachments/assets/ae780f17-f8a1-42e3-b76a-79ada727693a)

**Comment by charlesg99** [comment]
Working as intended:

![image](https://github.com/user-attachments/assets/61e76bfe-d6c9-4781-a4bd-42d6c56994ab)

**Comment by yioannides** [comment]
Yeah, but that hides the panel altogether, whereas I want to hide it when only there's a window open, regardless of the size, and have it on screen when there's no windows open.

**Comment by charlesg99** [comment]
I understood what you want :) 

What I'm saying is that the current options are working as intended even if they don't allow the behavior you want.

**Comment by yioannides** [comment]
Sorry, I must have phrased it poorly and thank you for editing the title! So if I understand correctly is that the "All windows" option hasn't been fully implemented? Otherwise, I am not sure what "All windows" currently does.

**Comment by charlesg99** [comment]
A picture (or two) is worth a thousand words

![image](https://github.com/user-attachments/assets/0f732dab-1ffd-4e19-95e4-102d205ade56)

![image](https://github.com/user-attachments/assets/bf013cfd-ad87-4658-805f-0af1cc516c6f)

Hope that makes sense!

**Comment by yioannides** [comment]
Hmmm, it's not working for me for some reason.

![image](https://github.com/user-attachments/assets/cbd50f2b-f9d3-48ff-a6eb-862823a7ffa7)

**Comment by deusnovus** [comment]
came here to request that as well, it would be a great feature to have !

**Comment by yioannides** [comment]
I gave it a try and it works perfectly! Thank you so much for your hard work! 😊

**Comment by yioannides** [comment]
There seems to be a bug occurring after `suspend` where Dash to Panel looks like this:

**On the main window:**
![Image](https://github.com/user-attachments/assets/5d365ceb-e116-49f3-a879-415c68e9e3de)

**On overview:**
![Image](https://github.com/user-attachments/assets/47faa243-c171-4877-8d13-2defcc201ade)

The Dash to Panel reverts back to its intended form after disabling + re-enabling the extension:
![Image](https://github.com/user-attachments/assets/4d7ceeeb-b7b7-4219-a42c-0f408a82f33b)

**Comment by charlesg99** [comment]
Thanks for testing! Can you give me more info about your system (gnome version, xorg or wayland, looks like Blur my Shell)? I can't reproduce the suspend issue so far. Do you normally see the Dash in the overview? If not, there probably is a js error somewhere that causes it to no be disabled by dtp.

**Comment by charlesg99** [comment]
I can't reproduce so far. Next time you come back from suspend and the issue occurs, could you please run `journalctl --since='20 minutes ago' > ~/dtp_log.txt` and send me the dtp_log.txt file from your home folder? Thanks!

**Comment by yioannides** [comment]
Yes of course, I produced a log file just now, with the past two minutes of action:

[dtp_log.txt](https://github.com/user-attachments/files/19705134/dtp_log.txt)

**Comment by charlesg99** [comment]
Thanks for the log, unfortunately I think the error happens when suspending and the log wasn't long enough. I see a bunch of errors, but those are the result of something being broken already. Anyway, while trying something else I noticed that I broke something on disable, not sure if this is what's causing the issue you're seing but there's a good chance it is. Can you please update to the latest master and try again?

**Comment by yioannides** [comment]
Ah thank you so much for checking it out! Unfortunately, it didn't fix this issue after updating to the latest master, and additionally crashed my GDM session once and introduced some visual glitches I haven't had before...

Everything happened after a clean restart, so I am attaching a new 20 minute log here:
[dtp_log.txt](https://github.com/user-attachments/files/19710259/dtp_log.txt)

**Comment by charlesg99** [comment]
Thanks, I don't see a dash to panel error that could cause the crash :( The only errors I see seem to come from rounded-window-corners@fxgn. Could you disable your other extensions, then log out/in and see if that helps? 

Also it's weird, the only dash to panel thing in your log is a warning that I already fixed on master, are you sure you're running the master branch (you can confirm that buy looking in the "About" tab, the commit hash you're running should be next to the version)? 

![Image](https://github.com/user-attachments/assets/e50ecc0c-4f34-460f-82aa-cb44b931307e)

**Comment by yioannides** [comment]
Okay, I seem to have found the root of the problem: it occurs when the `Blur My Shell` extension and `Dash To Panel > Settings > Override Panel Theme Background Opacity` are both turned on... as I mentioned before, the bug vanishes after disabling + re-enabling Dash To Panel.

I did a dconf dump of my extensions settings to see whether you may be able to reproduce my issue:
[extension-settings.txt](https://github.com/user-attachments/files/19712185/extension-settings.txt)

Thank you so much for your patience!

**Comment by charlesg99** [comment]
Yes you're right, not even needed to suspend, when "Override background" is enabled in the panel blur settings, disabling Blur my Shell crashes Gnome on my system. I'll look into that, thanks!

**Comment by charlesg99** [comment]
Added a workaround for the crash, looks like Blur my Shell is accessing the panel while it is being destroyed. This happens when Blur my Shell is loaded before Dash to Panel. In this case, when disabling Blur my Shell, Gnome first disables dtp then disables Blur my Shell, then re-enable dtp. Can you please test it out and see if that resolve the suspend issue?

**Comment by charlesg99** [comment]
Hmm sorry not sure what else, I could reproduce "a" crash but the one I had should be fixed. Does it only happen on suspend for you? I could reproduce it by:

- disable dtp and Blur my Shell
- restart gnome-shell
- enable Blur my Shell
- enable dtp
- disable Blur my Shell
- crash - "Oh no, something...."

If it does crash for you like this, can you please send me your logs again?

**Comment by 4JX** [comment]
Slightly hijacking the conversation that reopened the issue here since I've been experiencing the "half sized background" part of it for a while now, currently testing on `ad8c3ea`.

I can easily trigger the issue by reorganizing my workspaces via the [space bar](https://extensions.gnome.org/extension/5090/space-bar/) extension by dragging on the numbers. This is in combination with `dash-to-panel` and `blur-my-shell` which are the actually conflicting ones here.

![Image](https://github.com/user-attachments/assets/981f94c7-06b0-4631-a10e-f10ecc7a0b6e)

Once I trigger the issue, disabling `blur-my-shell` reveals the `dash-to-panel` stock background which seems to be properly sized. Re-enabling it afterwards brings the problem back. Disabling and reenabling `dash-to-panel` by itself does nothing as well.

###### There's a faint background there, which coincides with my config for `dtp` to play nicely with `bms`
![Image](https://github.com/user-attachments/assets/d915be32-4ef8-41b2-8eef-96b9876a4b63)

The only way to "fix" this is by disabling both and then loading `blur-my-shell` followed by `dash-to-panel`. `journalctl` captures no logs in any of this, save for some keybinding(?) related options for `dtp`.

I'm running a freshly upgraded Gnome 48 on Wayland setup, though this happened throughout some part of 47 as well. No shell reloads are needed to reproduce this, and I had no crashes in testing either.
